githubapp/middleware.go

```diff
@@ -118,16 +118,6 @@ func ClientLogging(lvl zerolog.Level) ClientMiddleware {
 	}
 }
 
-// ClientCaching creates client middleware that caches http responses
-// using the provided cache implementation
-func ClientCaching(cache func() httpcache.Cache) ClientMiddleware {
```

Comment:
Removing this is a breaking change, but its broken and misleading, which is arguably worse.